runtime/interpreter/encode.go

```diff
@@ -683,7 +684,8 @@ func (e *Encoder) prepareDictionaryValue(
 		return nil, err
 	}
 
-	entries := make(map[string]interface{}, v.Entries.Len())
+	// Use CBOR array for entry value to preserve ordering and improve speed.
+	entries := make([]interface{}, 0, v.Entries.Len())
```

Comment:
Could we use `cborArray` here and in other places?
```suggestion
	entries := make(cborArray, 0, v.Entries.Len())
```

Reply:
Yes!  Nice catch!  For this PR, I should do this in other places if it's in functions for dictionaries.  For non-dictionaries like composite types, I'd prefer to do it in the PR for composite types but I'm OK with either approach.

Reply:
sure, that sounds good!